src/helpers/pagination.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @class
+ * @classdesc - Pagination Class
+ */
+class Pagination {
+  /**
+   * @constructor
+   * @memberof Pagination
+   * @param {number|string} page
+   * @param {number|string} limit
+   */
+  constructor(page = 1, limit = 10) {
+    this.page = Number(page, 10);
+    this.limit = Number(limit, 10);
+  }
+
+  /**
+   * @memberof Pagination
+   * @param {number} page
+   * @return {object} - an object that returns query metadata
+   */
+  getQueryMetadata() {
+    return { limit: this.limit, offset: this.limit * (this.page - 1) };
+  }
+
+  /**
+   * @memberof Pagination
+   * @param {array} totalItems
+   * @param {string} baseUrl
+   * @param {string} extraQuery
+   * @return {object} - an object that returns page metadata
+   */
+  getPageMetadata(totalItems, baseUrl, extraQuery) {
+    const { page, limit } = this;
+
+    const pages = Math.ceil(totalItems / limit);
+
+    const prev =
```

Comment:
The logic here is a bit difficult to follow. I think splitting the individual sections into their own variables would reduce the complexity.

Also, use `null` instead of `undefined`. It is generally agreed to use `null` for when setting empty values instead of `undefined`

Reply:
All right, noted! thank you.